docs/core/settings/variables.md

```diff
@@ -27,24 +30,102 @@ You can use special variables in several places:
+
+Filters can accept parameters, both positional and named. E.g. `%{literal('\r\n\')}` will expand
+to CR LF. `%{user | substr(1)}` will take first character of username.
+
+You can use `%%{variable}` to escape this, and emit `%{variable}`.
+
+Filters accept strings, numbers and variables as parameters. Key names cannot be variables.
+The left side of pipe character (`|`) is provided as input to a filter. Some filters can be used
+in place of variables, e.g. lookup, literal and if.
+
+When value is missing or empty, you can use the `default` filter to provide value. Missing variables
+will cause errors and must be negated with default. This does not apply to all providers, some
+providers return empty when value is missing.
+
+The new syntax also supports simple maths, you can do one operation. E.g. `%{port + 1000}`.
+Addition, substraction, multiplication, division and modulo operations are supported for now.
+
+All strings must be encapsulated with " or ', and you can escape them using \ within string.
+Numbers when used as parameters must be provided without quotes.
+
+## List of filters
+
+All parameters are strings unless stated otherwise.
+If parameter is any, it accepts both numbers and strings.
+Boolean type is 0 = false or 1 = true.
+
+| Filter | Description |
+| ------ | ----------- |
+| lookup(name) | Lookup var from table. If var is variable, the name is taken from variable's contents. |
+| literal(string) | Expands into literally the value. If variable is used, works like lookup. |
+| concat(any, any...) | Concatenates input with value(s). Numbers are coerced to strings. Input is optional. |
+| upper | Uppercases input. |
+| lower | Lowercases input. |
+| hash(method, rounds=number, salt=string) | Returns raw hash from input using given hash method. Rounds and salt are optional. |
+| md5(rounds=number, salt=string) | Alias for hash with method md5. |
+| sha1(rounds=number, salt=string) | Alias for hash with method sha1. |
+| sha256(rounds=number, salt=string) | Alias for hash with method sha256. |
+| sha384(rounds=number, salt=string) | Alias for hash with method sha384. |
+| sha512(rounds=number, salt=string) | Alias for hash with method sha512. |
+| base64(pad=boolean, url=boolean) | Base64 encode given input, defaults to pad and not url scheme. |
+| unbase64(pad=boolean, url=boolean) | Base64 decode given input, defaults to pad and not url scheme. |
+| hex(number) | Hex encode input, with optional width. |
+| unhex | Hex decode input. |
+| default(value) | Replace empty or missing input with value. Clears missing variable error. If no value is provided, "" is used. |
+| reverse | Reverse input bytes. |
+| truncate(len, bits=number) | Truncate to len bytes, or number of bits. |
+| substr(start, end) | Substring bytes of input from start to end. Start and end can be negative numbers, in which case they are counted from right. End is optional. |
+| ldap_dn | Converts domain.com to dc=domain,dc=com. |
```

Comment:
\` escaping would be good all over the place in this table when referring to parameters or example values.